fodselsnummer.py

```diff
@@ -27,24 +27,54 @@ def check_fnr(fnr, d_numbers=True):
     if not FNR_REGEX.match(fnr):
         return False
 
-    try:
-        datetime.strptime(fnr[0:6], '%d%m%y')
-    except ValueError:
-        if d_numbers:
-            dnrdatestring = str(int(fnr[0])-4) + fnr[1:6]
-            try:
-                datetime.strptime(dnrdatestring, '%d%m%y')
-            except ValueError:
-                return False
+    individual_number = int(fnr[6:9])
+    day, month, year = int(fnr[0:2]), int(fnr[2:4]), int(fnr[4:6])
+    if 41 <= day <= 71:  # if D-number
+        if not d_numbers:
+            logger('Day out of range in fnr')
```

Comment:
Maybe we should state real reason of rejecting fnr, e.g.: "fodselsnumber is a D-number"